mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/feedback/SendFeedbackViewModel.kt

```diff
@@ -0,0 +1,113 @@
+
+    private val _feedbackSavedToBeSubmitted = SingleLiveEvent<Boolean>()
+    val feedbackSavedToBeSubmitted: LiveData<Boolean>
+        get() = _feedbackSavedToBeSubmitted
+
+    private val _progress = MutableLiveData<Boolean>()
+    val progress: LiveData<Boolean> get() = _progress
+
+    private val _draftFeedBackInstance = MutableLiveData<FeedbackInstance>()
+    val draftFeedBackInstance: LiveData<FeedbackInstance> get() = _draftFeedBackInstance
+
+    @SuppressLint("CheckResult")
+    fun submitFeedback(instance: FeedbackInstance) {
+        disposables.add(feedbackRepository.submitFeedback(instance).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).doOnSubscribe { _progress.postValue(true) }.doFinally { _progress.postValue(false) }.subscribe({
+            it.apply {
+                _feedbackSubmitted.postValue(true)
+            }
+
+        }) { throwable: Throwable? ->
+            Timber.d(throwable)
+            if (throwable is NoConnectivityException) {
+                _feedbackSubmitted.postValue(false)
+            } else {
+                FirebaseCrashlytics.getInstance().recordException(throwable!!)
+            }
+        })
+
+    }
+
+
+    fun saveFeedbackToBeSubmitted(feedbackInstance: FeedbackInstance) {
+        disposables.add(dataSource.saveInstance(feedbackInstance).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).doOnSubscribe { _progress.postValue(true) }.doFinally { _progress.postValue(false) }.subscribe({
+            it.apply {
+                _feedbackSavedToBeSubmitted.postValue(true)
+            }
+
+        }) { throwable: Throwable? ->
+            Timber.d(throwable)
+            if (throwable is NoConnectivityException) {
+                _feedbackSavedToBeSubmitted.postValue(false)
+            } else {
+                FirebaseCrashlytics.getInstance().recordException(throwable!!)
+            }
+        })
+
+    }
+
+    fun saveFeedbackDraft(feedbackInstance: FeedbackInstance) {
+        disposables.add(dataSource.saveInstance(feedbackInstance).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).doOnSubscribe { _progress.postValue(true) }.doFinally { _progress.postValue(false) }.subscribe({
+            it.apply {
+                _feedbackSavedAsDraft.postValue(true)
+            }
+
+        }) { throwable: Throwable? ->
+            Timber.d(throwable)
+            FirebaseCrashlytics.getInstance().recordException(throwable!!)
+        })
+
+    }
+
+    fun getFeedBackDraft() {
+        disposables.add(dataSource.getFeedbackDraft().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).doOnSubscribe { _progress.postValue(true) }.doFinally { _progress.postValue(false) }.subscribe({ draft ->
+            _draftFeedBackInstance.postValue(draft)
+
+        }) { throwable: Throwable? ->
+            Timber.d(throwable)
+            FirebaseCrashlytics.getInstance().recordException(throwable!!)
```

Comment:
I don't think so because it is very rare and we don't have anything about it on figma. I asked dhekra and she didn't handle it but I can ask Caro too about it.